Add playlist link support

dgw opened the pull request:
Example output:

```irc
<dgw> https://www.youtube.com/playlist?list=PLMmqC65vzdRkFt8T0QSy8GuCYMV8nfodU
<SopelTest> [YouTube] Factorio 1.0 Gameplay | Playlist by ImKibitz | 3 items | Created Tuesday, August 18, 2020 13:46:27 (UTC)
```

I elected not to implement output customization for this because:

1. There aren't enough info items about playlists to really matter
2. I don't want to deal with the `info_items` config value not specifying that it's for videos, in any way (don't want to deal with renaming it either).

dgw committed:
Rename some stuff to be "video" specific

Making way for playlist support.

dgw committed:
Handle playlist links

Does not fetch and display playlist description because it's almost
certainly going to be MUCH too long for IRC display, or empty; there's
not much in between those two extremes.

About the only useful information we can get about playlists is the
title, channel name, item count, and creation time. YouTube's API does
not expose a playlist's last-update time anywhere, sadly.

cottongin reviewed (approved):
Looks good to me, even works with direct video links that have the playlist bit attached (I'm assuming ignoring the playlist and returning the direct video info is intended behavior)

e.g. [https://www.youtube.com/watch?v=bPRfTKiMbEI&list=PLMmqC65vzdRkFt8T0QSy8GuCYMV8nfodU&index=2&t=0s](https://www.youtube.com/watch?v=bPRfTKiMbEI&list=PLMmqC65vzdRkFt8T0QSy8GuCYMV8nfodU&index=2&t=0s)
```
[YouTube] I've Waited YEARS to Play This! - Factorio 1.0 Gameplay Ep 1 | 24m 5s | Channel: ImKibitz | 129,045 views | 7,078+ | 240- | 2020-08-18 17:00
```

dgw committed:
Show playlist info too for watch links with a list ID

Can be disabled with a new setting.

dgw commented:
@cottongin I actually just didn't think about it. Since it was easy to do, I just did (with a setting, just in case). Can always lop off that last commit before merging 🤪

dgw committed:
playlist handler silently ignores Watch Later ID & falsy (empty) IDs

Ignoring None, 0, '', etc. is for saving on useless API requests, and
for avoiding exceptions from trying to call `.upper()` on a non-string.

Ignoring the WL playlist is for saving on useless requests too, since
YouTube's API won't return any data about it unless there's an
authenticated user associated with the request (since Watch Later is
stored on a per-user basis).
